tests/api2/test_040_ad_user_group_cache.py

```diff
@@ -1,165 +1,127 @@
-    with active_directory(
-        AD_DOMAIN,
-        ADUSERNAME,
-        ADPASSWORD,
-        netbiosname=hostname,
-        createcomputer=AD_COMPUTER_OU,
-    ) as ad:
-        yield (request, ad)
+    with active_directory() as ad:
+        # make sure we are exra sure cache fill complete
+        cache_fill_job = call(
+            'core.get_jobs',
+            [['method', '=', 'directoryservices.cache.refresh']],
+            {'order_by': ['-id'], 'get': True}
+        )
+        if cache_fill_job['state'] == 'RUNNING':
+            call('core.job_wait', cache_fill_job['id'], job=True)
 
+        yield ad
 
-@pytest.mark.dependency(name="AD_IS_HEALTHY")
-def test_03_enabling_activedirectory(do_ad_connection):
-    global WORKGROUP
 
-    results = GET('/activedirectory/started/')
-    assert results.status_code == 200, results.text
+def get_ad_user_and_group(ad_connection):
+    WORKGROUP = ad_connection['dc_info']['Pre-Win2k Domain']
 
-    results = GET("/smb")
-    assert results.status_code == 200, results.text
-    WORKGROUP = results.json()['workgroup']
+    domain_prefix = f'{WORKGROUP.upper()}{WINBIND_SEPARATOR}'
+    ad_user = ad_connection['user_obj']['pw_name']
+    ad_group = f'{domain_prefix}domain users'
 
+    user = call(
+        'user.query', [['username', '=', ad_user]],
+        {'extra': {'search_dscache': True}, 'get': True}
+    )
 
-@pytest.mark.dependency(name="INITIAL_CACHE_FILL")
-def test_06_wait_for_cache_fill(request):
-    """
-    Local user/group cache fill is a backgrounded task.
-    Wait for it to successfully complete.
-    """
-    depends(request, ["AD_IS_HEALTHY"])
-    results = GET(f'/core/get_jobs/?method=activedirectory.fill_cache')
-    job_status = wait_on_job(results.json()[-1]['id'], 180)
-    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    group = call(
+        'group.query', [['name', '=', ad_group]],
+        {'extra': {'search_dscache': True}, 'get': True}
+    )
+
+    return (user, group)
 
 
-@pytest.mark.dependency(name="AD_USERS_CACHED")
-def test_07_check_for_ad_users(request):
+def test_check_for_ad_users(do_ad_connection):
     """
     This test validates that we can query AD users using
     filter-option {"extra": {"search_dscache": True}}
     """
-    depends(request, ["INITIAL_CACHE_FILL"], scope="session")
     cmd = "wbinfo -u"
     results = SSH_TEST(cmd, user, password)
     assert results['result'], str(results['output'])
-    wbinfo_entries = results['stdout'].splitlines()
+    wbinfo_entries = set(results['stdout'].splitlines())
 
-    results = GET('/user', payload={
-        'query-filters': [['local', '=', False]],
-        'query-options': {'extra': {"search_dscache": True}},
-    })
-    assert results.status_code == 200, results.text
-    assert len(results.json()) > 0, results.text
-    cache_names = [x['username'] for x in results.json()]
+    cache_names = set([x['username'] for x in call(
+       'user.query',
```

Comment:
More vision problems.   Indentation is only 3.
This same indentation issue is repeated at lines: 82, 108, 116, 142, 150, 186 and 194.